Warnings C4244 and C4305 'argument': truncation / conversion from 'double' to 'RawType' / 'const M' to 'const Element'

compiled with Visual Studio 2019 Professional 16.5.1 on windows 10 1903.

`\src\benchmark_runner.cc(267,13): warning C4244: 'initializing': conversion from 'double' to 'benchmark::IterationCount', possible loss of data
\src\benchmark_runner.cc(266,41): warning C4244: 'initializing': conversion from 'double' to 'const benchmark::IterationCount', possible loss of data
\test\statistics_gtest.cc(25,1): warning C4305: 'argument': truncation from 'double' to 'RawType' \test\statistics_gtest.cc(24,1): warning C4244: 'argument': conversion from 'double' to 'RawType', possible loss of data
\test\../src/benchmark_register.h(25,1): warning C4244: '*=': conversion from 'int' to 'T', possible loss of data
\src\googlemock\include\gmock/gmock-matchers.h(368,12): warning C4244: 'argument': conversion from 'const M' to 'const Element', possible loss of data

--- a/src/benchmark_runner.cc
+++ b/src/benchmark_runner.cc
@@ -263,8 +263,9 @@ class BenchmarkRunner {
     if (multiplier <= 1.0) multiplier = 2.0;
 
     // So what seems to be the sufficiently-large iteration count? Round up.
-    const IterationCount max_next_iters =
-        std::lround(std::max(multiplier * i.iters, i.iters + 1.0));
+    const IterationCount max_next_iters = static_cast<IterationCount>(
+        std::lround(std::max(multiplier * static_cast<double>(i.iters),
+                             static_cast<double>(i.iters) + 1.0)));
     // But we do have *some* sanity limits though..
     const IterationCount next_iters = std::min(max_next_iters, kMaxIterations);
 

--- a/test/statistics_gtest.cc
+++ b/test/statistics_gtest.cc
@@ -22,7 +22,7 @@ TEST(StatisticsTest, StdDev) {
   EXPECT_DOUBLE_EQ(benchmark::StatisticsStdDev({101, 101, 101, 101}), 0.0);
   EXPECT_DOUBLE_EQ(benchmark::StatisticsStdDev({1, 2, 3}), 1.0);
   EXPECT_DOUBLE_EQ(benchmark::StatisticsStdDev({2.5, 2.4, 3.3, 4.2, 5.1}),
-                   1.1510864433221342);
+                   1.151086443322134);
 }
 
 }  // end namespace

--- a/test/statistics_gtest.cc
+++ b/test/statistics_gtest.cc
@@ -21,8 +21,8 @@ TEST(StatisticsTest, Median) {
 TEST(StatisticsTest, StdDev) {
   EXPECT_DOUBLE_EQ(benchmark::StatisticsStdDev({101, 101, 101, 101}), 0.0);
   EXPECT_DOUBLE_EQ(benchmark::StatisticsStdDev({1, 2, 3}), 1.0);
-  EXPECT_FLOAT_EQ(benchmark::StatisticsStdDev({1.5, 2.4, 3.3, 4.2, 5.1}),
-                  1.42302495);
+  EXPECT_DOUBLE_EQ(benchmark::StatisticsStdDev({2.5, 2.4, 3.3, 4.2, 5.1}),
+                   1.1510864433221342);
 }
 
 }  // end namespace

